pkg/dashboard/resource/v3iostream.go

```diff
@@ -113,6 +199,127 @@ func (vsr *v3ioStreamResource) getNamespaceFromRequest(request *http.Request) st
 	return vsr.getNamespaceOrDefault(request.Header.Get("x-nuclio-project-namespace"))
 }
 
+func (vsr *v3ioStreamResource) getShardLagsMap(ctx context.Context, info v3ioStreamInfo) (map[string]restful.Attributes, error) {
+
+	// create v3io context
+	v3ioContext, err := v3iohttp.NewContext(vsr.Logger, &v3iohttp.NewContextInput{
+
+		// by default client id nil
+		HTTPClient: vsr.V3ioHTTPClient,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed creating v3io context")
+	}
+
+	// a data plane access key is required for v3io operations
+	accessKey := vsr.getCtxSession(ctx).GetPassword()
+	if accessKey == "" {
+		return nil, errors.Wrap(err, "A data plane access key is required")
+	}
+
+	dataPlaneInput := v3io.DataPlaneInput{
+		URL:           vsr.getPlatform().GetConfig().StreamMonitoring.WebapiURL,
+		AccessKey:     accessKey,
+		ContainerName: info.ContainerName,
+	}
+	getContainerContentsInput := &v3io.GetContainerContentsInput{
+		Path:             info.StreamPath,
+		GetAllAttributes: true,
+		DirectoriesOnly:  false,
+		DataPlaneInput:   dataPlaneInput,
+	}
+
+	shardLagsMap := map[string]restful.Attributes{}
+	lock := sync.Mutex{}
+
+	for {
```

Comment:
add comments why a for / what is happening within and etc. be concise, provide meaningful comments for developers / readers